api/admin/controller.py

```diff
@@ -1101,30 +1158,54 @@ def admin_auth_services(self):
             else:
                 return NO_PROVIDER_FOR_NEW_ADMIN_AUTH_SERVICE
 
-        # Only Google OAuth is supported for now.
-        url = flask.request.form.get("url")
-        username = flask.request.form.get("username")
-        password = flask.request.form.get("password")
-        domains = flask.request.form.get("domains")
+        if provider == AdminAuthenticationService.GOOGLE_OAUTH:
+            url = flask.request.form.get("url")
+            username = flask.request.form.get("username")
+            password = flask.request.form.get("password")
+            domains = flask.request.form.get("domains")
         
-        if not url or not username or not password or not domains:
-            # If an admin auth service was created, make sure it
-            # isn't saved in a incomplete state.
-            self._db.rollback()
-            return INCOMPLETE_ADMIN_AUTH_SERVICE_CONFIGURATION
+            if not url or not username or not password or not domains:
+                # If an admin auth service was created, make sure it
+                # isn't saved in a incomplete state.
+                self._db.rollback()
+                return INCOMPLETE_ADMIN_AUTH_SERVICE_CONFIGURATION
 
-        # Also make sure the domain list is valid JSON.
-        try:
-            json.loads(domains)
-        except Exception:
-            self._db.rollback()
-            return INVALID_ADMIN_AUTH_DOMAIN_LIST
-
-        integration = auth_service.external_integration
-        integration.url = url
-        integration.username = username
-        integration.password = password
-        integration.set_setting("domains", domains)
+            # Also make sure the domain list is valid JSON.
+            try:
+                json.loads(domains)
+            except Exception:
+                self._db.rollback()
+                return INVALID_ADMIN_AUTH_DOMAIN_LIST
+
+            integration = auth_service.external_integration
+            integration.url = url
+            integration.username = username
+            integration.password = password
+            integration.set_setting("domains", domains)
+
+        elif provider == AdminAuthenticationService.LOCAL_PASSWORD:
+            admins = flask.request.form.get("admins")
+
+            if not admins:
+                self._db.rollback()
+                return INCOMPLETE_ADMIN_AUTH_SERVICE_CONFIGURATION
+
+            # Also make sure admins is valid JSON.
+            try:
+                admins = json.loads(admins)
+            except Exception:
+                self._db.rollback()
+                return INVALID_ADMIN_AUTH_ADMINS_LIST
+
+            for admin_details in admins:
+                admin, ignore = get_one_or_create(self._db, Admin, email=admin_details.get("email"))
+                if (admin_details.get("password")):
+                    admin.password = admin_details.get("password")
```

Comment:
Any newly created admins will have a password here. Admins that already existed will only have an email.